### ISSUE TITLE
Reveal / Hide Debug GUI

### ISSUE THREAD

**Issue by REAS** [processing/processing-experimental#9]
We need to start thinking about how to hide and reveal the menu icons needed to debug a program. Will this be a visual part of the PDE, will it be in the text menus, etc. Do you have ideas?

**Comment by Manindra29** [comment]
How about we restore the toolbar to Java mode style, and when user starts the debugger(from Mode menu), a separate window pops up with the debug buttons/variable inspector? Have the debug part show up only when users wants it.

**Comment by REAS** [comment]
Yes, I think something like this is the right thing to do. Definitely the first part, as we discussed with @shiffman, it should start like Java mode menu. And it makes sense to only have the "debug" console there when a user wants it. I'm not sure about opening a separate window, but I don't have a good alternative right now.

**Comment by aengelke** [comment]
Couldn't it go to the console/errors area if it doesn't go in another window? It could look like this (just an idea, buttons are continue, step over, step into, step out (?)):

![debug](https://f.cloud.github.com/assets/4236689/1022250/2bbf9b90-0d74-11e3-8989-ea32f59f0965.png)

**Comment by Manindra29** [comment]
Okay, this can be an alternative approach, instead of opening another window. Adding all the debug buttons and the variable inspector as a separate pane in the bottom panel. But it would also increase the button count in the line status bar, which I think is cluttered a bit already. What are your thoughts @REAS, @shiffman ?

**Comment by Manindra29** [comment]
In fact, I just figured out a way to switch between Java Mode toolbar and Debugger toolbar on the fly, without restarting the editor. I added a toggle option in the Debug menu. So, the user can now switch between both toolbars as he requires. I think this settles it?

**Comment by aengelke** [comment]
A solution would be to only show the debug buttons, when a breakpoint is hit. But I personally don't like jumping toolbar buttons or a mix of general (open, save, etc.) buttons (this would probably the case if you put the debug buttons at the end of the toolbar).

**Comment by Manindra29** [comment]
Just to be clear, this is how they appear:

Normal:
![s22](https://f.cloud.github.com/assets/1686425/1022611/19d195d0-0d9e-11e3-84ac-91dceb89fca7.png)

Debugger:
![s11](https://f.cloud.github.com/assets/1686425/1022612/1a0ef2c2-0d9e-11e3-96f6-c1fa4536dfa8.png)

**Comment by shiffman** [comment]
A toggle option in the debug menu seems great.  Do the breakpoints also toggle?  I think that extra column should be hidden when the debugger is not enabled.  In addition, I think the mode should probably start with the debugger turned off.  Ideally it might remember how it was last viewed?

One last idea is instead of having the option in the debug menu, to include an extra button that just turns the debugger on or off.  Maybe that button could live next to where it says "PDE X".

**Comment by Manindra29** [comment]
It should be possible to toggle the breakpoints column too. I  have turned off the debugger by default in the repo already. I'll have to see about saving the current view..

Adding an extra button doesn't seem right. I think we should stick the original PDE ui as much as we can. We could rename the Debug menu to something else maybe like "Advanced" or "PDE X"?

**Comment by shiffman** [comment]
Good points, let's leave it as a toggle option in the menu for now and see how that goes.  I'm not sure about the naming, if the "Debug" menu is only for debugging then maybe that is right.  If it involves other PDEX settings then maybe something more generic?

**Comment by Manindra29** [comment]
Yes, other than the Debug options, it has controls for many PDE X settings too. Should I rename it to 'PDE X'? Since the Debugger as well as the control options come under PDE X.

**Comment by shiffman** [comment]
Sure, for now, let's rename it, but we'll need to come up with a final name should it eventually become the main PDE and not PDE X!

**Comment by Manindra29** [comment]
Breakpoints now hidden on toggling Debugger, 27e44b9d0101524f3f9fe28c48c431192fbd42d2
